subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/metadata/PublishArtifactLocalArtifactMetadataSerializer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.metadata;
+
+import org.gradle.api.artifacts.PublishArtifact;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.ComponentIdentifierSerializer;
+import org.gradle.api.internal.artifacts.publish.ImmutablePublishArtifact;
+import org.gradle.internal.component.local.model.PublishArtifactLocalArtifactMetadata;
+import org.gradle.internal.serialize.Decoder;
+import org.gradle.internal.serialize.Encoder;
+import org.gradle.internal.serialize.Serializer;
+
+import java.io.File;
+
+public class PublishArtifactLocalArtifactMetadataSerializer implements Serializer<PublishArtifactLocalArtifactMetadata> {
+
+    private final ComponentIdentifierSerializer componentIdentifierSerializer;
+
+    public PublishArtifactLocalArtifactMetadataSerializer(ComponentIdentifierSerializer componentIdentifierSerializer) {
+        this.componentIdentifierSerializer = componentIdentifierSerializer;
+    }
+
+    @Override
+    public PublishArtifactLocalArtifactMetadata read(Decoder decoder) throws Exception {
+        ComponentIdentifier identifier = componentIdentifierSerializer.read(decoder);
+        PublishArtifact publishArtifact = new ImmutablePublishArtifact(
+            decoder.readString(),
```

Comment:
💄 given the lack of named parameter syntax in java I'd introduce a variable for each decoder call result so the `read` and `write` method implementations can appear more symmetric.

Reply:
Done and I did the same on all the introduced serializers.